addon/authenticators/devise.js

```diff
@@ -150,5 +150,16 @@ export default BaseAuthenticator.extend({
     }, options || {});
 
     return $.ajax(requestOptions);
+  },
+
+  _validate(data) {
+    const tokenAttributeName = this.get('tokenAttributeName');
+    const identificationAttributeName = this.get('identificationAttributeName');
+    const resourceName = this.get('resourceName');
+    if (data[resourceName]) {
+      return !isEmpty(data[resourceName][tokenAttributeName]) && !isEmpty(data[resourceName][identificationAttributeName]);
+    } else {
+      return !isEmpty(data[tokenAttributeName]) && !isEmpty(data[identificationAttributeName]);
+    }
```

Comment:
I think this would be simpler as

```js
const resourceName = this.get('resourceName');
const _response = response[resourceName] ? response[resourceName] : response;

return !isEmpty(response[tokenAttributeName]) && !isEmpty(response[identificationAttributeName]);
```